scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -511,26 +543,45 @@ class PropertySpec:
       non-alphanumeric characters to underscores.
 
     const:
-      The property's constant value as given in the binding, or None.
+      The property's constant value as given in the binding, or None. It can
+      be inherited from a matching pattern property.
 
     default:
       The property's default value as given in the binding, or None.
 
     deprecated:
-      True if the property is deprecated; False otherwise.
+      True if the property is deprecated; False otherwise. It can be inherited
+      from a matching pattern property. It can be inherited from a matching
+      pattern property.
 
     required:
-      True if the property is marked required; False otherwise.
+      True if the property is marked required; False otherwise. It can be
+      inherited from a matching pattern property.
 
     specifier_space:
-      The specifier space for the property as given in the binding, or None.
+      The specifier space for the property as given in the binding, or None. It
+      can be inherited from a matching pattern property.
+
+    pattern:
+      A compiled regular expression for the property's pattern, or None.
     """
 
     def __init__(self, name: str, binding: Binding):
         self.binding: Binding = binding
         self.name: str = name
         self._raw: Dict[str, Any] = self.binding.raw["properties"][name]
 
+        self.type = self._raw.get("type", "")
+        self.enum = self._raw.get("enum")
+        self.const = self._raw.get("const")
+        self.deprecated = self._raw.get("deprecated", False)
+        self.required = self._raw.get("required", False)
+        self.specifier_space = self._raw.get("specifier-space")
```

Comment:
Dropped types that used to be attached to property handlers.
```suggestion
        self.type: str = self._raw.get("type", "")
        self.enum: Optional[list] = self._raw.get("enum")
        self.const: Union[None, int, List[int], str, List[str]] = self._raw.get("const")
        self.deprecated: bool = self._raw.get("deprecated", False)
        self.required: bool = self._raw.get("required", False)
        self.specifier_space: Optional[str] = self._raw.get("specifier-space")
```